src/main/java/org/opensearch/plugin/insights/core/service/TopQueriesService.java

```diff
@@ -267,6 +321,51 @@ public List<SearchQueryRecord> getTopQueriesRecords(final boolean includeLastWin
         if (includeLastWindow) {
             queries.addAll(topQueriesHistorySnapshot.get());
         }
+        List<SearchQueryRecord> filterQueries = queries;
+        if (from != null && to != null) {
+            final DateTime start = DateTime.parse(from);
+            final DateTime end = DateTime.parse(to);
+            Predicate<SearchQueryRecord> timeFilter = element -> start.getMillis() <= element.getTimestamp()
+                && element.getTimestamp() <= end.getMillis();
+            filterQueries = queries.stream().filter(removeInternal.and(timeFilter)).collect(Collectors.toList());
```

Comment:
We are applying the filter to remove the search requests to internal indices for in-memory records only. Should we also apply the same filter for historical data in index?

Reply:
I found that the internal records didn't seem to be stored in the historical data, but I can put in the filter as well just in case.